test/unit/test_row_factory.py

```diff
@@ -0,0 +1,54 @@
+from unittest import mock
+import pytest
+from etlhelper.row_factories import namedtuple_rowfactory
+import sqlite3
+
+
+# Test uses sqlite3 cursor, but all dbapi compliant cursors will be the same.
+mock_cursor = mock.MagicMock(sqlite3.Cursor)
+mock_cursor.fetchmany.return_value = [
+    (1, 2, 3),
+    (4, 5, 6),
+]
+
+
+def test_valid_field_names():
+    mock_cursor.description = (('id', None, None, None, None, None, None),
+                               ('name', None, None, None, None, None, None),
+                               ('desc', None, None, None, None, None, None))
+
+    create_row = namedtuple_rowfactory(mock_cursor)
+
+    rows = (create_row(row) for row in mock_cursor.fetchmany(50))
+
+    for i, row in enumerate(rows):
+        assert row._fields[0] == "id"
```

Comment:
This can be a one-liner

`assert row._fields == ("id", "name", "desc)`

There are a few examples of this below.